### ISSUE TITLE
request:  Add `init` command to bootstrap a melos project

### ISSUE THREAD

**Issue by exaby73** [invertase/melos#773]
### Is there an existing feature request for this?

- [X] I have searched the existing issues.

### Command

melos init

### Description

Running `melos init` should in theory automate the steps defined in [Getting Started][getting-started]

[getting-started]: https://melos.invertase.dev/getting-started

### Reasoning

Currently setting up a melos project is very manual having to create your `pubspec.yaml` and `melos.yaml`. This new `init` command would solve that

**Comment by spydon** [comment]
Should I assign you to it @exaby73? :)

**Comment by exaby73** [comment]
I got it :) I'll cook up something soon thank you

**Comment by albinpk** [comment]
I've made a Mason brick to bootstrap a Melos project structure with poly-repo support, maybe useful while the `melos init` command is in progress.
https://brickhub.dev/bricks/melos

**Comment by spydon** [comment]
There is also this brick: https://github.com/ubuntu/flutter_project_template

But we don't want to depend on mason for the init command.

**Comment by spydon** [comment]
@albinpk ah now I get the comment, you meant as a work-around until `melos init` is out? That's a good idea. :)

**Comment by albinpk** [comment]
> @albinpk ah now I get the comment, you meant as a work-around until `melos init` is out? That's a good idea. :)

Yes, exactly; it's a workaround until `melos init` is available.

@spydon  I’m actually using Melos in my smaller projects without a monorepo structure as well, since Melos is really useful for versioning, tagging, and changelog generation. I set it up with a simple `melos.yaml` like this:

```yaml
name: <name>

packages:
  - ./

command:
  version:
    workspaceChangelog: false
```

Is there another recommended way to set up Melos for projects that aren’t monorepos, or could we add an option for this setup in the `init` command?

**Comment by spydon** [comment]
It isn't recommended to have a project directly in the root, because the dependencies of the project (and vice versa) then need to conform to the Melos version and its dependency constraints.
So even with only one app/package I would recommend to use a subdirectory.

It would be good if this could be done in a better manner, but not sure how that would be possible (and that would be related to another issue and not this one).

**Comment by albinpk** [comment]
Ya, that makes sense about the dependencies. Thanks for the clarification!

**Comment by JHubi1** [comment]
To be conformed with other popular CLI applications, like Flutter and the Dart CLI itself, the command should rather be `melos create` right?
